MapboxNavigationTests/HistoricRouteProgressTests.swift

```diff
@@ -0,0 +1,30 @@
+import XCTest
+import FBSnapshotTestCase
+import Turf
+import TestHelper
+import MapboxDirections
+@testable import MapboxCoreNavigation
+@testable import MapboxNavigation
+
+
+class HistoricRouteProgressTests: FBSnapshotTestCase {
+
+    override func setUp() {
+        super.setUp()
+        recordMode = false
+        agnosticOptions = []
+    }
+
+    func testHistoricRouteProgress() {
+        // Short initial route
+        let route = Fixture.route(from: "historic-route-progress")
+        // Trace of a detour from the initial origin to the destination
+        let trace = Fixture.locations(from: "historic-route-progress.trace")
```

Comment:
For the sake of future efforts, how was this trace generated?

Reply:
`trace` was generated by a detour version of the `route` (same origin and destination but including a third detour waypoint) using `Fixture.generateTrace(for:)`, will add to the comment above.